tests/python/relay/test_tidl.py

```diff
@@ -0,0 +1,210 @@
+    assert model_compile(model_name, relay_mod, relay_params, data_layout, input_node, input_data)
+
+
+def test_tidl_object_detection(model_name):
+    im_fname = download_testdata('https://github.com/dmlc/web-data/blob/master/' +
+                                 'gluoncv/detection/street_small.jpg?raw=true',
+                                 'street_small.jpg', module='data')
+    image_size = 512
+    data_layout = "NCHW"
+    dtype = "float32"
+    input_name = "data"
+
+    #======================== Load testing image and model ====================
+    input_shape = (1, 3, image_size, image_size)
+    x, img = data.transforms.presets.ssd.load_test(im_fname, short=image_size)
+    block, ssd_mod, ssd_params = load_gluoncv_model(model_name, x, input_name, input_shape, dtype)
+
+    #======================== Execute the full graph on TVM ====================
+    with relay.build_config(opt_level=3):
+        graph, lib, params = relay.build(ssd_mod, "llvm", params=ssd_params)
+    mod = graph_runtime.create(graph, lib, ctx=tvm.cpu(0))
+    input_data = x.asnumpy()
+    np.save("ssd_input.npy",input_data) # to be used by inference testing on the target
+    mod.set_input(input_name, input_data)
+    mod.set_input(**params)
+    mod.run()
+    class_IDs, scores, bounding_boxs = mod.get_output(0), mod.get_output(1), mod.get_output(2)
+    ax = utils.viz.plot_bbox(img, bounding_boxs.asnumpy()[0], scores.asnumpy()[0],
+                             class_IDs.asnumpy()[0], class_names=block.classes)
+    plt.savefig("gluoncv_"+model_name+"_tvm.png")
+    results = [mod.get_output(i).asnumpy() for i in range(mod.get_num_outputs())]
+    print("Number of outputs: " + str(len(results)))
+
+    #======================== Compile the model ========================
+    assert model_compile(model_name, ssd_mod, ssd_params, data_layout, input_name, input_data)
+
+def test_tidl_segmentation(model_name):
+    input_name = "data"
+    data_layout = "NCHW"
+    dtype = "float32"
+
+    #======================== Load testing image and model ====================
+    img_url = 'https://raw.githubusercontent.com/dmlc/web-data/master/gluoncv/segmentation/voc_examples/1.jpg'
+    filename = 'example.jpg'
+    utils.download(img_url, filename)
+    img = image.imread(filename)
+    img = test_transform(img, ctx = mx.cpu(0))
+    input_shape = img.shape
+    input_data = img.asnumpy()
+    np.save("seg_input.npy",input_data) # to be used by inference testing on the target
+    model = model_zoo.get_model(model_name, pretrained=True)
+    model.hybridize()
+    model.forward(img)
+    model.export('gluoncv-temp')    # create file gluoncv-temp-symbol.json
+    seg_mod, seg_params = relay.frontend.from_mxnet(model, {input_name:input_shape})
+
+    #======================== Compile the model ========================
+    assert model_compile(model_name, seg_mod, seg_params, data_layout, input_name, input_data)
+
+if __name__ == '__main__':

```

Comment:
Can you move the loops over model lists into the tests themselves? For example, instead of  
```
if __name__ == "__main__":
  classification_models   = ['mobilenet1.0', 'mobilenetv2_1.0']
  for model in classification_models:
    test_tidl_classification(model)
  # ...
```

in __main__ just do:
```
if __name__ == "__main__":
  test_tidl_classification()
  # ...
```

and then `test_tidl_classification()` will have the loop:
```
def test_tidl_classification():
  for model in ['mobilenet1.0',  'mobilenetv2_1.0']:
    # current code from test_tidl_classification here
```

Reply:
Done